### PR TITLE
py-igor: new package

### PR DESCRIPTION
**Pull request by heerener**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-igor/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    """igor: interface for reading binary IGOR files."""
+
+    homepage = "http://blog.tremily.us/posts/igor/"
+    pypi = "igor/igor-0.3.tar.gz"
+
+    version("0.3", sha256="b04ac38c68fb81cf3167a9981dc5a20379112d40268bb72c5a8514dc8051abba")
+
+    def patch(self):
+        setupfile = FileFilter("setup.py")
+        setupfile.filter("distutils.core", "setuptools")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor/package.py]: I believe pip does this automatically. Not sure about build which we're planning to switch to in the future.

File: var/spack/repos/builtin/packages/py-igor/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    depends_on("py-setuptools", type="build")
+    depends_on("py-numpy", type="run")
+    depends_on("py-nose", type="run")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor/package.py]: Nose is only needed for unit tests, which we don't run here. I would remove it.

File: var/spack/repos/builtin/packages/py-igor/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    depends_on("py-matplotlib", type="run")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-igor/package.py]: I would make numpy and matplotlib `type=("build", "run")` so the import tests pass.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by heerener** [comment]
@adamjstewart is that docs build something I can fix from my end?

**Comment by adamjstewart** [comment]
Probably transient, let me try kicking the tests

**Pull request merged**